core/chain.py

```
@@ -0,0 +1,287 @@
+            raise ValueError("LLM client must have either 'achat' or 'chat' method")
+
+        return result
+
+    def _build_llm_input(self, context: ChainContext) -> str:
+        """构建 LLM 输入"""
+        # 格式化 prompt 使用上下文数据
+        formatted_prompt = self.prompt
+        try:
+            formatted_prompt = self.prompt.format(**context.data)
+        except (KeyError, ValueError) as e:
+            # 如果格式化失败，使用原始 prompt
```

Ruff (0.15.10)

[warning] Comment contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF003)